columnq/src/table/database.rs

```diff
@@ -28,125 +25,15 @@ mod imp {
-                            _ => tokio_postgres::NoTls,
-                        };
-                        let source: PostgresSource<
-                            postgres::BinaryProtocol,
-                            tokio_postgres::NoTls,
-                        > = PostgresSource::new(config.into(), tls, 2)
-                            .map_err(|e| ColumnQError::Database(e.to_string()))?;
-                        let queries = queries.clone();
-                        let task = tokio::task::spawn_blocking(move || {
-                            let dispatcher = Dispatcher::<
-                                PostgresSource<postgres::BinaryProtocol, tokio_postgres::NoTls>,
-                                ArrowDestination,
-                                PostgresArrowTransport<
-                                    postgres::BinaryProtocol,
-                                    tokio_postgres::NoTls,
-                                >,
-                            >::new(
-                                source, &mut destination, &queries, None
-                            );
-
-                            if let Err(e) = dispatcher.run() {
-                                return Err(ColumnQError::Database(e.to_string()));
-                            }
-
-                            let schema_ref = destination.arrow_schema();
-                            match destination.arrow() {
-                                Ok(data) => datafusion::datasource::MemTable::try_new(
-                                    schema_ref,
-                                    vec![data],
-                                )
-                                .map_err(|e| ColumnQError::Database(e.to_string())),
-                                Err(e) => Err(ColumnQError::Database(e.to_string())),
-                            }
-                        });
-
-                        // FIXME: Maybe use other way to block the async task instead of block_on
-                        futures::executor::block_on(task)
-                            .map_err(|e| ColumnQError::Database(e.to_string()))?
-                    }
-                    #[cfg(not(feature = "database-postgres"))]
-                    {
-                        return Err(ColumnQError::Database(
-                            "Postgres database feature not enabled.".to_string(),
-                        ));
-                    }
-                }
-            }
+            let queries = CXQuery::naked(format!("SELECT * FROM {}", t.name));
+            let source = SourceConn::try_from(t.get_uri_str())
+                .map_err(|e| ColumnQError::Database(e.to_string()))?;
+            let destination = connectorx::get_arrow::get_arrow(&source, None, &[queries])
+                .map_err(|e| ColumnQError::Database(e.to_string()))?;
+            Ok(datafusion::datasource::MemTable::try_new(
+                destination.arrow_schema(),
+                vec![destination.arrow().unwrap()],
```

Comment:
instead of doing an unwrap here, we should propagate the error to the caller instead so we don't crash the server

Reply:
Yes thank you!  I will make another PR to fix this, since the previous PR has already been merged.